src/AbstractAlgebra.jl

```diff
@@ -110,6 +110,142 @@ export error_dim_negative, ErrorConstrDimMismatch
 
 export crt
 
+###############################################################################
+# Macros for fancy printing and extending objects when desired
+# fancy printing (and extending)
+# change
+#
+# struct bla..
+# ..
+# end
+# 
+# to 
+#
+# struct bla ..
+# @declare_other
+# ...
+# end
+#
+# Then, in the show function, start with
+# @show_name(io, obj)
+# If the user assigned a name to the object (in the REPL mainly) by doing
+# A = bla...
+# then, in the compact printing only the name "A" is printed
+# also adding
+# @show_special(io, obj)
+# allows, if present to call a different printing function for this instance
+# See FreeModule for an example
+#
+###############################################################################
+
+macro declare_other()
+   esc(quote other::Dict{Symbol, Any} end )
```

Comment:
A `Dict` is heavy-weight: for small objects which must be created cheaply, this could be a problem. As this is used apparenly only for printing, is it too expensive to lookup each time the name in `Main` without caching? For other uses of `other` besides the name, there might also be alternatives? (e.g for "different printing function for this instance", using a global dictionary).